torch/package/package_exporter.py

```diff
@@ -454,6 +466,11 @@ def close(self):
         if self.verbose:
             print(f"Dependency graph for exported package: \n{self._write_dep_graph()}")
 
+        # Check that all mock and extern modules with allow_empty=False were matched.
+        for i, (pattern, _, allow_empty) in enumerate(self.patterns):
+            if not allow_empty and i not in self.matched_patterns:
+                raise RuntimeError(f"Exporter did not match {pattern} to any modules")
```

Comment:
lgtm, but let's define a new class of errors (EmptyMatchError?) so that users can catch it systematically.

Reply:
And let's add some more explicit language tying this error to the choice of `allow_empty`, something like: we specified X pattern but it did not match any modules, this is an error because allow_empty=False.

Reply:
How should we print out `X pattern`? Right now it gets printed out using `_GlobGroup.__str__`, which produced something like `_GlobGroup(include=..., exclude=...)`. Would that make sense to someone using this?